simplemma/strategies/dictionaries/trie_directory_factory.py

```diff
@@ -0,0 +1,161 @@
+import hashlib
+import logging
+from collections.abc import MutableMapping
+from functools import lru_cache
+from pathlib import Path
+from typing import ByteString, Dict, List, Optional, cast
+
+from marisa_trie import BytesTrie, HUGE_CACHE  # type: ignore[import-not-found]
+from platformdirs import user_cache_dir
```

Comment:
`platformdirs` is also a external dependency. So, why it doesn't need the type ignore like `marisa_trie`?

Reply:
While I'm not entirely sure, my guess is that mypy can't get any type information, because marisa_trie isn't Python code, but a C-extension and doesn't provide type information.